runtime/gc_base/GCExtensions.cpp

```diff
@@ -240,8 +240,9 @@ MM_GCExtensions::updateIdentityHashDataForSaltIndex(uintptr_t index)
 }
 
 /*
- * computeDefaultMaxHeapForJava is for Java only, it will be called during gcParseCommandLineAndInitializeWithValues(),
- * computeDefaultMaxHeap() will still be called during MM_GCExtensionsBase::initialize(), computeDefaultMaxHeapForJava() can overwrite value of memoryMax.
+ * computeDefaultMaxHeapForJava() is used for Java only.
+ * computeDefaultMaxHeap() will still be called during MM_GCExtensionsBase::initialize().
+ * computeDefaultMaxHeapForJava() can overwrite value of memoryMax.
```

Comment:
`  * computeDefaultMaxHeapForJava() can overwrite value of memoryMax.`

I don't think this part is true either. Not due to your changes, but due to previous (Restore related) changes. This method used to directly change ext->memoryMax, but not anymore - it just calculates a value and the caller may/will change memoryMax.